README.md

```diff
@@ -1,40 +1,81 @@
 
-* [Chase-Lev Work-Stealing Deque](src/ws_deque.mli) Single-producer, multi-consumer dynamic-size double-ended queue (deque) (see [Dynamic circular work-stealing deque](https://dl.acm.org/doi/10.1145/1073970.1073974) and [Correct and efficient work-stealing for weak memory models](https://dl.acm.org/doi/abs/10.1145/2442516.2442524)). Ideal for throughput-focused scheduling using per-core work distribution. Note, [pop] and [steal] follow different ordering (respectively LIFO and FIFO) and have different linearization contraints.
+**In isolation* : a domain is said to run *in isolation* when it is the only domain to take steps. The other domains do nothing during that time even if they are in the middle of a method call. It is used to described properties, but not an actual domain behaviour.
 
-* [SPSC Queue](src/spsc_queue.mli) Simple single-producer single-consumer fixed-size queue. Thread-safe as long as at most one thread acts as producer and at most one as consumer at any single point in time.
+### How does it works ?
+The basic idea is that all operations on shared data use `atomics` variables.
 
-* [MPMC Relaxed Queue](src/mpmc_relaxed_queue.mli) Multi-producer, multi-consumer, fixed-size relaxed queue. Optimised for high number of threads. Not strictly FIFO. Note, it exposes two interfaces: a lockfree and a non-lockfree (albeit more practical) one. See the mli for details. 
+### Why use lock free data structures ?
+#### In brief
+| Pros |  Cons |
+|------|-------|
+| thread safe | **not** starvation free |
+| non blocking and dead-lock safe |  not easily composable (see [reagents](https://github.com/ocaml-multicore/reagents)) |
+| usually good performances as it runs in parallel most often than blocking data structures | still subject to contention issues |
 
-* [MPSC Queue](src/mpsc_queue.mli) A multi-producer, single-consumer, thread-safe queue without support for cancellation. This makes a good data structure for a scheduler's run queue. It is used in [Eio](https://github.com/ocaml-multicore/eio). It is a single consumer version of the queue described in [Implementing lock-free queues](https://people.cs.pitt.edu/~jacklange/teaching/cs2510-f12/papers/implementing_lock_free.pdf).
+#### What does that means ?
+If you need a *non-blocking* data structure, you are at the right place. However, if you only need a *thread safe* one you may not find what you need here. Especially, lock free data structures does not necessary have better performances than blocking ones. However,  it ensures that is a thread is delayed, that won't delay the other ones (this is the obstruction freedom property)
 
-## Usage
+Also lock freedom is a restraining property, which means it is not easy (and it may not even be possible), to modify or add a function to a lock free data structure while maintaining this property. That explains why most of time the provided data structure are a very simple API.
```

Comment:
```suggestion
Also lock freedom is a restraining property, which means it is not easy (and it may not even be possible), to modify or add a function to a lock free data structure while maintaining this property. That explains why most of time the provided data structures have a very simple API.
```